planetiler-core/src/main/java/com/onthegomap/planetiler/reader/ShapefileReader.java

```diff
@@ -150,34 +132,30 @@ private ShapefileDataStore open(Path path) {
   }
 
   @Override
-  public long getCount() {
+  public long getFeatureCount() {
     return inputSource.size();
   }
 
   @Override
-  public WorkerPipeline.SourceStep<SimpleFeature> read() {
-    return next -> {
-      try (var iter = inputSource.features()) {
-        long id = 0;
-        while (iter.hasNext()) {
-          id++;
-          org.opengis.feature.simple.SimpleFeature feature = iter.next();
-          Geometry source = (Geometry) feature.getDefaultGeometry();
-          Geometry latLonGeometry = source;
-          if (transformToLatLon != null) {
-            latLonGeometry = JTS.transform(source, transformToLatLon);
-          }
-          if (latLonGeometry != null) {
-            SimpleFeature geom = SimpleFeature.create(latLonGeometry, new HashMap<>(attributeNames.length), sourceName,
-              null, id);
-            for (int i = 1; i < attributeNames.length; i++) {
-              geom.setTag(attributeNames[i], feature.getAttribute(i));
-            }
-            next.accept(geom);
+  public void readFeatures(Consumer<SimpleFeature> next) throws TransformException {
+    try (var iter = inputSource.features()) {
+      while (iter.hasNext()) {
+        org.opengis.feature.simple.SimpleFeature feature = iter.next();
+        Geometry source = (Geometry) feature.getDefaultGeometry();
+        Geometry latLonGeometry = source;
+        if (transformToLatLon != null) {
+          latLonGeometry = JTS.transform(source, transformToLatLon);
+        }
+        if (latLonGeometry != null) {
+          SimpleFeature geom = SimpleFeature.create(latLonGeometry, new HashMap<>(attributeNames.length),
+            sourceName, null, featureId.incrementAndGet());
```

Comment:
I don't think this ID needs to be globally unique, would probably be fine to just increment a local counter here to convey what order it was contained in the file.

Reply:
Is it okay if different features in the same source layer (potentially also the same tile) have the same id?

Reply:
There's a different process that generates vector tile feature IDs. The only use of this ID is if you need conditional logic to handle a specific feature differently and all you have to go by is the order that it appears in an input file.